sphinx/application/data/features/feature-media-cache/src/main/java/io/matthewnelson/feature_media_cache/MediaCacheHandlerImpl.kt

```diff
@@ -51,12 +54,30 @@ class MediaCacheHandlerImpl(
         }
     }
 
+    private val paidTextCache: File by lazy {
+        File(cacheDir, PAID_TEXT_CACHE_DIR).also {
+            it.mkdirs()
+        }
+    }
+
     override fun createImageFile(extension: String): File =
         createFileImpl(imageCache, IMG, extension)
 
     override fun createVideoFile(extension: String): File =
         createFileImpl(videoCache, VID, extension)
 
+    override fun createPaidTextFile(): File {
+        if (!paidTextCache.exists()) {
+            synchronized(cacheDirLock) {
+                if (!paidTextCache.exists()) {
+                    paidTextCache.mkdirs()
+                }
+            }
+        }
+
+        return File(paidTextCache, PAID_TEXT_FILE)
```

Comment:
do not use static file name. Network lagginess can result in overwritting of the file content before it gets uploaded to meme server, or worse, overwritten during the upload which would crash the app.

use `createFileImpl(paidTextCache, TXT, "txt")` to create a unique file name for the message.